libs/sdk-model/src/insight/tests/insight.test.ts

```diff
@@ -548,3 +550,32 @@ describe("insightModifyItems", () => {
         );
     });
 });
+
+describe("insightReduceItems", () => {
+    const modifications: BucketItemReductions = (
+        acc: IAttributeOrMeasure[],
+        cur: IAttributeOrMeasure,
+        idx: number,
+        _src: IAttributeOrMeasure[],
+    ): IAttributeOrMeasure[] => {
+        // reduce to just one attribute for bucket
+        if (isAttribute(cur) && idx > 0) {
+            return [cur];
+        }
+        return [...acc, cur];
+    };
+
+    it("should return another insight with empty bucket", () => {
+        const modifiedInsight: IInsight = insightReduceItems(EmptyInsight, modifications);
+        expect(modifiedInsight !== EmptyInsight).toBe(true);
+        expect(modifiedInsight.insight.buckets.length).toBe(0);
+    });
+
+    it("should return another insight with reduced bucket items and only one attribute in bucket", () => {
+        const insightWithSomeBuckets: IInsight = insightSetBuckets(EmptyInsight, [MultiAttributeBucket]);
+
+        const modifiedInsight: IInsight = insightReduceItems(insightWithSomeBuckets, modifications);
+        expect(modifiedInsight !== insightWithSomeBuckets).toBe(true);
```

Comment:
Please use `expect(modifiedInsight).not.toEqual(insightWithSomeBuckets)`, it will provide much nicer output on failure. This version will just say something along the lines of `Expected "false" to equal "true"`

Reply:
Fixed